modules/dreamview/backend/teleop/teleop.cc

```diff
@@ -104,23 +132,84 @@ void TeleopService::RegisterMessageHandlers() {
       });
 }
 
-void TeleopService::UpdateModemInfo(
-    const std::shared_ptr<ModemInfo> &modem_info) {
-  // TODO simplify data and only send necessary info for display
-  if (modem_info->has_provider() && modem_info->has_technology()) {
-    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
-    teleop_status_["modems"][modem_info->provider()] = modem_info->technology();
-  }
-}
 
 void TeleopService::SendStatus(WebSocketHandler::Connection *conn) {
   std::string to_send;
   {
     boost::shared_lock<boost::shared_mutex> reader_lock(mutex_);
     to_send = teleop_status_.dump();
+    AINFO << "status: " << to_send << std::endl;
   }
   websocket_->SendData(conn, to_send);
 }
 
+
+void TeleopService::UpdateModem(unsigned int index,
+    const std::shared_ptr<ModemInfo> &modem_info) {
+
+    // AINFO << "UpdateModem: " << index << std::endl;
+    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+    // TODO simplify data and only send necessary info for display
+    // update modem_info_
+    if (modem_info->has_provider() && modem_info->has_technology()) {
+        boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+        std::string modemId = "0";
+        if (index == 0) {
+            modemId = "1";
+        }
+        if (index == 1) {
+            modemId = "2";
+        }
+        if (index == 2) {
+            modemId = "3";
+        }
+
+        // teleop_status_["modems"][modem_info->provider()] =
+        //  modem_info->technology();
+        teleop_status_["modems"][modemId] = modem_info->technology();
+    }
+}
+
+void TeleopService::UpdateCarDaemonRpt(
+    const std::shared_ptr<DaemonServiceRpt> &daemon_rpt) {
+  {
+      boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+      // AINFO << "CarDaemonRpt" << std::endl;
+      bool aVideoEncoderIsRunning = false;
+      bool voipIsRunning = false;
+      for (int i = 0; i < daemon_rpt->services_size(); i++) {
+          std::string service =  daemon_rpt->services(i);
+          AINFO <<  "  *" << service << std::endl;
+          if (service.find("encoder") >= 0) {
+              aVideoEncoderIsRunning = true;
+          }
+          if (service.find("voip_encoder") >= 0) {
+              voipIsRunning = true;
+          }
+      }
+      teleop_status_["video"] = aVideoEncoderIsRunning;
+      teleop_status_["audio"] = voipIsRunning;
+  }
+}
+
+void TeleopService::UpdateOperatorDaemonRpt(
+    const std::shared_ptr<DaemonServiceRpt> &daemon_rpt) {
+  {
+      boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+      // AINFO << "OperatorDaemonRpt" << std::endl;
+      bool voipIsRunning = false;
+      for (int i = 0; i < daemon_rpt->services_size(); i++) {
+          std::string service =  daemon_rpt->services(i);
+          AINFO <<  "  *" << service << std::endl;
+          if (service.find("voip_encoder") >= 0) {
+              voipIsRunning = true;
+          }
+      }
+      teleop_status_["mic"] = voipIsRunning;
```

Comment:
if this means the report of operator side daemon outcome when remotely requested by car, then voipIsRunning should set both mic and audio?

Reply:
I admit I did not think about the car requesting the operator daemon outcome, and 
we were only thinking about the operator's perspective for the UI.  We don't even relay the operator daemon's report to the car. But we can change it.

Reply:
If we don't consider the car requesting the operator daemon outcome, then why do we need two readers for daemon rpt and two callback functions handling them separately? I am ok with just showing car daemon outcome.

Reply:
The car's daemon is used to start / stop / check the video encoding and the audio encoding in the car. The operator daemon is used to start / stop /check the audio in the operator computer.  

Without the operator daemon, controlling the operator voip encoder from the operator console computer would need to be done with a different mechanism or a separate UI (the task manager).

Reply:
I understand we need two daemons working on each side. But here is about the reporting status from each daemon. In your previous comment, you mentioned there will not be operator daemon status reported to the car, which is ok for me. If that is the case, we only need one reader to subscribe to the car daemon status report, right?

Reply:
Without the operator daemon report / command, how should we stop/start and check the status the voip_encoder in the local (operator) machine? This seamed like the solution to that problem.

Reply:
> I admit I did not think about the car requesting the operator daemon outcome, and
> we were only thinking about the operator's perspective for the UI. We don't even relay the operator daemon's report to the car. But we can change it.

All my discussion about not needing the report status of operator daemon is based on this comment. If we do have both daemons reporting its status to the remote side, then we just need to set the appropriate teleop_status_ fields correctly. I did not want to remove the report status of operator daemon in the first place. Also, have a daemon running vs. reporting its status are two separate issues, right? I was just saying that if you did not think about reporting the operator daemon outcome while it is indeed running, it is ok with me.